dataland-frontend/tests/sharedUtils/components/UploadDocuments.ts

```diff
@@ -53,4 +53,14 @@ export const uploadReports = {
     cy.get(`[data-test="${reportName}ToUploadContainer"]`).should("not.exist");
     cy.get(`[data-test="${reportName}AlreadyUploadedContainer"]`).should("not.exist");
   },
+  assertUploadButtonDisappeared(): void {
+    cy.get('button[data-test="upload-files-button"]').should("not.exist");
+  },
+
+  selectDocumentAtEachFileSelector(filename: string): void {
```

Comment:
Does this really select a document at every selector?
Iam asking from a performance point of view, how long did the test run and were there memoriy problems?

Reply:
Work on this test is still in progress so this might change slightly. But it looks like we need to select a document for each selector since the test requires a dataset with all "Yes" selected. This leads to all possible file selectors to be filled to pass validation.

Reply:
Maybe this will lead to crashes of the cypress testrunner, because the input gets to big.

Reply:
locally it did work with the same "hacky trick" of using the browser window and providing all documents in this way, waiting now for the CI.

Reply:
done

Reply:
CI is now only red because of two code smells induced by two false-negative keycloak usages

Reply:
as discussed, these will be set to ignore by the reviewer in sonar

Reply:
Taking a new look into the code